apps/meteor/client/lib/chats/data.ts

```diff
@@ -247,7 +249,17 @@ export const createDataAPI = ({ rid, tmid }: { rid: IRoom['_id']; tmid: IMessage
 
 	const isSubscribedToRoom = async (): Promise<boolean> => !!subscriptionsCollection.findOne({ rid }, { reactive: false });
 
-	const joinRoom = async (): Promise<void> => call('joinRoom', rid);
+	const joinRoom = async (): Promise<void> => {
+		const room = await findRoom();
+		if (!room) {
+			throw new Error('Room not found');
+		}
+		if (isRoomFederated(room)) {
+			await APIClient.post('/v1/federation/joinInternalPublicRoom', { internalRoomId: rid });
+			return;
+		}
+		await call('joinRoom', rid);
+	};
```

Comment:
Can this be done server side?
I'm concerned there may be a situation where the join function is called while the room is not yet loaded into the cached collection - this new check would prevent the user from joining the room.

Reply:
For sure, I decided to do this since in the other PR I was removing the support for the `afterAddedUserToRoom` event, but I added it back, so now we can rely again on the joinRoom method, I'll fix that.